lib/grammar/rules/assignment_word.cpp

```diff
@@ -0,0 +1,67 @@
+#include "shell/grammar/rules/assignment_word.hpp"
+#include "shell/interfaces/provider.hpp"
+#include "shell/assert.hpp"
+#include "shell/grammar/grammar_util.hpp"
+
+namespace shell {
+
+Rule AssignmentWord::make() {
+	return Rule {
+	    AssignmentWord::options,
+	};
+}
+
+vector<Rule::Option> AssignmentWord::options() {
+	return {Rule::Terminal {handler}};
+}
+
+bool AssignmentWord::isValidName(const string &input) {
+	if (input.empty()) {
+		return false;
+	}
+	auto first_char = input[0];
+	if (isdigit(first_char)) {
+		return false;
+	}
+	for (auto &ch : input) {
+		auto is_underscore = ch == '_';
+		auto is_digit = isdigit(ch);
+		auto is_alpha = isalpha(ch);
+		if (!is_underscore && is_digit && is_alpha) {
```

Comment:
This conditions seems off, I think it should be `!is_underscore && !is_digit && !is_alpha`

Reply:
You're right, messed that up 😅